### ISSUE TITLE
Issue setting up Fission with Kubernetes

### ISSUE THREAD

**Issue by Wang-Siyuan** [fission/fission#135]
Hi there,

We are trying to use Fission framework as part of a research project at school.  When setting up Fission on top of Kubernetes, we ran into the following issue after following the instructions on your repo:

root@f240f6d3795b:/# kubectl --namespace fission get svc
NAME         CLUSTER-IP       EXTERNAL-IP   PORT(S)        AGE
controller   10.104.16.147    <nodes>       80:31313/TCP   8s
etcd         10.105.208.189   <none>        2379/TCP       14s
poolmgr      10.107.138.3     <none>        80/TCP         14s
router       10.104.182.227   <nodes>       80:31314/TCP   8s
root@f240f6d3795b:/# export FISSION_URL=http://10.104.16.147:31313
root@f240f6d3795b:/# export FISSION_ROUTER=10.104.182.227:31314
root@f240f6d3795b:/# fission env create --name nodejs --image fission/node-env
Failed to create environment: Post http://10.104.16.147:31313/v1/environments: dial tcp 10.104.16.147:31313: i/o timeout

Are we not setting up FISSION URL and FISSION_ROUTER address properly? Any help is much appreciated. Thank you!

**Comment by soamvasani** [comment]
1. Check that the controller is up: `kubectl --namespace fission get pod`; look at the STATUS column.

If it's not `Running`, check for errors/logs with `kubectl --namespace fission describe <podname>` and  `kubectl --namespace fission logs <podname>`

2. Check that you can reach the FISSION_URL: `curl http://$FISSION_URL/`

**Comment by soamvasani** [comment]
Ah, you're using the wrong IP address -- the "CLUSTER-IP" field is not accessible from outside the cluster.  If you set up the services with `fission-nodeport.yaml`, you'll need to use the node's IP address in the fission URL, with the port.

**Comment by soamvasani** [comment]
Closing this.  @Wang-Siyuan please reach out if you need any other help, or if the above comment doesn't solve the problem for you.

**Comment by Wang-Siyuan** [comment]
Hi Soam,

Thanks so much for the help. I tried using the node's IP address but still couldn't get create the environment.

Here is the node's address:
`root@0edef74ad229:/# kubectl --namespace fission describe nodes
Name:                   0edef74ad229
Role:                   master
Labels:                 beta.kubernetes.io/arch=amd64
                        beta.kubernetes.io/os=linux
                        kubeadm.alpha.kubernetes.io/role=master
                        kubernetes.io/hostname=0edef74ad229
Taints:                 dedicated=master:NoSchedule
CreationTimestamp:      Tue, 07 Mar 2017 17:21:36 +0000
Phase:
Conditions:
  Type                  Status  LastHeartbeatTime                       LastTransitionTime                      Reason                          Message
  ----                  ------  -----------------                       ------------------                      ------                          -------
  OutOfDisk             False   Tue, 07 Mar 2017 17:31:57 +0000         Tue, 07 Mar 2017 17:21:36 +0000         KubeletHasSufficientDisk        kubelet has sufficient disk space available
  MemoryPressure        False   Tue, 07 Mar 2017 17:31:57 +0000         Tue, 07 Mar 2017 17:21:36 +0000         KubeletHasSufficientMemory      kubelet has sufficient memory available
  DiskPressure          False   Tue, 07 Mar 2017 17:31:57 +0000         Tue, 07 Mar 2017 17:21:36 +0000         KubeletHasNoDiskPressure        kubelet has no disk pressure
  Ready                 True    Tue, 07 Mar 2017 17:31:57 +0000         Tue, 07 Mar 2017 17:21:36 +0000         KubeletReady                    kubelet is posting ready status
Addresses:              172.17.0.1,172.17.0.1,0edef74ad229`

According to Kubernetes's documentation, the second entry in the last line indicates the external IP address of the node.

But setting it to fission controller/router URL did not help.

In fact, I am not seeing controller and router listening on the localhost at all(I only have one node right now and it's the localhost):
`root@0edef74ad229:/# nmap -sT -O localhost

Starting Nmap 7.01 ( https://nmap.org ) at 2017-03-07 17:34 UTC
Nmap scan report for localhost (127.0.0.1)
Host is up (0.000044s latency).
Other addresses for localhost (not scanned): ::1
Not shown: 998 closed ports
PORT     STATE SERVICE
8080/tcp open  http-proxy
9898/tcp open  monkeycom`

So It looks like it's definitely not listening on port 31313/31314.

I have executed these two commands, which I thought should have deployed controller/router on these ports:
`kubectl create -f http://fission.io/fission.yaml
kubectl create -f http://fission.io/fission-nodeport.yaml`
